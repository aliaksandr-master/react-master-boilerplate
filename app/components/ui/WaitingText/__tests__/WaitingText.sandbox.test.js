/* eslint-env browser, jest */
/* eslint-disable react/jsx-no-bind */

import React from 'react';
import renderer from 'react-test-renderer';
import { wrapSandbox } from '../../../../vendor/react-sandbox/sandbox-staging';
import sandboxTrackerIcon from '../WaitingText.sandbox';
import Test from '../../../test/Test';



describe('rendering', () => {
  wrapSandbox(sandboxTrackerIcon).forEach(({ name, template, props }) => {
    it(`should render sandbox "${name}"`, () => {
      const TrackerIconSandboxComponent = template();

      const component = renderer.create(
        <Test>
          <TrackerIconSandboxComponent {...props} />
        </Test>
      );

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
