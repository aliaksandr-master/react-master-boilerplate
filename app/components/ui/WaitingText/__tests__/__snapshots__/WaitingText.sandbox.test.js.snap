// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render sandbox "block" 1`] = `
<span
  className="WaitingText WaitingText--themeBlock"
  style={
    Object {
      "height": 300,
    }
  }
>
  <span
    className="WaitingText__in"
  >
    <span
      className="WaitingText__text"
    >
      Loading
    </span>
    <span
      className="WaitingText__dot"
    />
    <span
      className="WaitingText__dot"
    />
    <span
      className="WaitingText__dot"
    />
  </span>
</span>
`;

exports[`rendering should render sandbox "default" 1`] = `
<span
  className="WaitingText"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <span
    className="WaitingText__in"
  >
    <span
      className="WaitingText__text"
    >
      Loading
    </span>
    <span
      className="WaitingText__dot"
    />
    <span
      className="WaitingText__dot"
    />
    <span
      className="WaitingText__dot"
    />
  </span>
</span>
`;
